### DISCUSSION TITLE
IOB not visible in pebble endpoint with current xDrip+ build

### DISCUSSION THREAD

**Discussion by gdcwatchfaces** [NightscoutFoundation/xDrip#3795] [Bug reports]
I have been using an old build of xDrip+ without issues.  
![unnamed](https://github.com/user-attachments/assets/152bb74e-f7f8-48d0-a03c-df15a0de4e84) 

This build properly identifies the IOB from the Omnipod 5 companion app

![unnamed](https://github.com/user-attachments/assets/b436350e-78e2-4b5f-b194-ddb7eff7507d)

Every time I am prompted to update the xDrip+ app the IOB stops working

![unnamed](https://github.com/user-attachments/assets/96be0421-48a3-4d82-9211-183be367789d)
Upon install of this version an error is received

![unnamed](https://github.com/user-attachments/assets/2723a291-cd47-47b5-bf81-6c4c260d52d3)

With the API turned OFF this is what is sent to the pebble endpoint

![unnamed](https://github.com/user-attachments/assets/7062c9fb-b67e-415f-a9c3-431ef2ba0b6a)

With the Fetch IOB from Companion App is turned Off this is what is returned (as expected)
![unnamed](https://github.com/user-attachments/assets/a847035e-7d16-4bed-95d6-de75cf562be0)

Why is this important?

![unnamed](https://github.com/user-attachments/assets/e64b9f9c-7c7f-4b20-ae13-419f4838347d)

Because I have migrated to the Eversense 365 CGM I need to use the new and improved xDrip+ Sync feature (Master/Follower) so that I can see values on devices that do not have the "native" apps installed on the mobile device.   Unlike Dexcom and Libre there does not seem to be much development.  I am able with the use of ESEL to get the Glucose from the 365 app to xDrip.  I would like to use the xDrip+ Sync feature so that my wife can be a "follower" to be able to see my glucose and if possible the IOB values on her  phone and in turn on her watch with the use of GluocDataHandler.  I would also like to be able to see these values on my tablet.

![unnamed](https://github.com/user-attachments/assets/cb00491e-b1f2-40df-ae2c-5bdd79be60aa)

Devices Used
Samsung Galaxy S24 Ultra
![Screenshot_20241204_113739_Settings](https://github.com/user-attachments/assets/df16781e-7259-4353-8e69-f6a34d8b1d3c)

Samsung Flip 6
![Screenshot_20241204_113831_Settings](https://github.com/user-attachments/assets/c3cd3359-0df5-4c00-b0c9-0f9af379038d)

Samsung Tab S9 FE+
![Screenshot_20241204_114007_Settings](https://github.com/user-attachments/assets/1b12dc84-e0a3-4172-a7ad-9492bcf53a40)

Any assistance and notification of a resolution would be greatly appreciated
gdcwatchfaces@gmail.com

**Comment by gdc27health** [comment]
Hello?
How can I check on the status of this issue?

IOB is kind of important to those of us using the Omnipod 5

Thanks in advance

**Reply by Navid200** [reply]
I'm looking at the screenshots you have provided and see too much information!
I see symbols of watches.
I have no way of understanding exactly what the problem is.

xDrip does a lot.  Not everyone knows everything there is about xDrip.  I certainly don't.  I try to figure out what a problem is and categorize it and label it accordingly so that the right person can find it.

If xDrip does not work on a phone, that becomes a serious problem.
If xDrip works fine on a phone but does not on a watch, it becomes a secondary problem as not everyone uses a watch.
I cannot figure out if the problem you are describing a problem with the watch only or not.

**Comment by gdcwatchfaces** [comment]
Navid
The problem its well identified....
DISREGARD the watch if that confuses you.

xDRIP lost the ability to send the IOB to the pebble endpoint.
Do you see all of that in the well documented bug report?

The april version works just fine
EVERY time you send an update and i update the data is gone

I have show you screenshots of the setting and screenshots from the pebble
endpoin

Let me know please i  have to explain more to you.

**Reply by Navid200** [reply]
What is pebble end point?
Under what conditions does it become necessary to use it?

**Comment by gdcwatchfaces** [comment]
I want to use the xDrip+ Sync capabilities.  To do that i need a build post
end of July.
If i install the current build as you see noted in the bug report.

IOB does NOT work.

Do you understand the issue now?

This is repeatable and reporducatble

**Reply by Navid200** [reply]
If you make personal attacks, I will stop corresponding with you.  Then, you will have to wait even longer.
My questions may bother you.  Believe me, I am not having a picnic here either.
I need to understand what the problem is.  Focus on the subject instead of focusing on me.

One more time state something like "do you understand ..." and you can ask someone else for help

**Comment by gdcwatchfaces** [comment]
WOW...

MY GOD!!!!  asking for clarification about understanding was taken WAY WAY
out of context.  My point was you seemed to be focused on the Watch issue.
I amvery very sorry you took it that way.  The intent was to show that the
watch is not the root of the problem. Yes I completely understand the
complexity of all of the different apps that diabetics must use to get the
data where we want it.  Fortunately xDrip plays a key role in getting data
available to the devices we desire.

The root of the problem appears to be that the Pebble endpoint is not being
updated.  If you refer back to (Discussion #2728) you will see that you
developed the Fetch IOB functionality and that this data is passed to the
pebble endpoint  http://127.0.0.1:17580/pebble

In the bug report I sent you I have shown that in the april build the data
is there.

If I went back and attempted a build from the end of July and the most
recent September release.  Neither of them pass the data to the pebble
endpoint http://127.0.0.1:17580/pebble

I hope that this clarification help you more clearly see what the issue
appears to be.

I'm happy to help in testing and providing additional details

**Reply by Navid200** [reply]
Can you identify two consecutive Nightly builds where the first works and the second does not?
Then, I can look at the diff between the two hoping to see what chance could have caused the problem.

**Comment by gdc27health** [comment]
Navid....

I can tell you this..

Version 645e8b4-2024.04.19  Works just great.

I tried a version both
2024723 - e6cb8d6.apk
20240917 - dfcbe80.apk

In the bug report I show after installing 20240917 - dfcbe80.apk
an error after installation that it has a bug
There are 2 screenshots of the data from the pebble endpoint
1 with the Fetch on and it returns a 0
the second when the Fetch is off returns unknown (that's expected)

How far back would I need to go to identify this issue and when it broke?

**Reply by Navid200** [reply]
Look at this: https://navid200.github.io/xDrip/docs/ReleaseNotes.html  
It shows you all the releases with a few words for each I have added to help see what is changed in each.

If I understand you correctly: 
2024.04.19 works
2024.09.17 fails
There are so many releases between those two.

I cannot test them all.  Even if I wanted to, I don't have the environment to see fail and pass.  You do.
Please identify 2 *consecutive* releases (no other releases between them) where the first one works and the second one fails.

In other words, I need you to identify two releases, not three.
And I need those two releases to be consecutive so there should be no releases between them.
And the newer one should be the very first release that fails because you have confirmed that the other one that was released just prior to the newer one works fine.

**Comment by gdc27health** [comment]
Navid....
Here is what i was able to do to identify the issue

I downloaded all builds from 20240419 to 20240830

Method
1. Install the version of the app that I know works 20240419
2. Configure the app manually.
3. Save Settings
4. Perform an update by installing the next build
I installed 9 builds ending in 20240628 that functioned as expected
The next build 20240717 failed.  It returned the value of "unknown" in the
endpoint

Between 20240628 and 20240717 something changed.


I went a step further.
I uninstalled 20240717 and attempted to go backwards with the builds
I attempted to install 20240628, configure manually and it failed.
I attempted to install 20240625, configure manually and it failed
I attempted to install 20240611, configured manually and it also failed

I went back to the known build 20240419 and it installs and functions as
intended

I did notice one thing if the user attempts to "restore" from previous
setting saved
In inter-app settings the Fetch is set to on.  However if the slider is
turned to the off position the user is prompted to allow further
notifications.  This is not related to the initial issue.  This is just
something I saw when attempting to install the build that i know works.

If there is any further testing setting or anything you need please do not
hesitate to ask for more.  I am a retired ERP systems implementation
consultant.  I am very familiar with working with development to duplicate,
replicate and document software issues.

Stay in range 70-180
Graham

**Reply by Navid200** [reply]
The fact that going between the two releases in one direction versus the other produces different outcomes suggests that you have done something inconsistently.

The only differences between those two releases are related to Dexcom G7 and G6 and some translation updates.

I suggest you install the version that works.
Do not uninstall.  Just install the next release.   There will be no need to restore any settings as long as you don't uninstall.

I am hoping you can identify two consecutive releases say A and B.

A should be working fine.  When you install B over A without an uninstall, it should fail.
Installing A over B should fix the problem again.
You should be able to go back and forth between A and B with the same results consistently.

**Comment by gdc27health** [comment]
Yes but there are not sequential builds that allow that.
The 0717 build fails
The build that reliiably installs is the 0419

Id love to get to a build that uses xDrip Sync

**Reply by Navid200** [reply]
> Yes but there are not sequential builds that allow that. 

What do you mean?
You can install any version of xDrip over any other version of xDrip with no need to uninstall.  

If you install B over A and it causes it to fail.  Then, you install A over B and it still fails, you have not found the two consecutive builds that one fails the other works.  It sounds like you are causing problems with settings as you restore settings.
If you install a version of xDrip over another, you will not need to change settings.

**Comment by gdcwatchfaces** [comment]
Navid...

I am going to stay on task and only identify the steps taken to get the IoB
Fetch feature to not function as expected.


IOB testing steps

1. All instances of xDrip and related files deleted
2. Mobile device restarted
3. xDrip build 2024041914 installed

4. DexShare configured with user id / password and us server
5. Settings > Inter-app stettings

   - Broadcast locally = ON
   - Noise Blocking > Send even Extremely noisy signals
   - Compatible Broadcast = checked
   - Identify receiver = de.michelinside.glucodatahandler
   - xDrip Web Service = ON
   - Broadcast Service API = ON
   - Fetch IoB from companion a.. = ON
   - Please Allow Permission = OK
   - Notification access slider to ON
   - Allow xDrip+ = Allow


6. Mobile device restarted
7.  127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
8. Confirm notification from Omnipod 5 app
9. Download & Install next sequential build 20240512-db904ab
10.Update
11. Open
12. Please Allow Permission = OK
13. Stop optimizing battery usage = Allow
14. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
15. Confirm notification from Omnipod 5 app
16. Download & Install next sequential build 20240515-c967ee7
17. Update
18. Open
19. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
20. Confirm notification from Omnipod 5 app
21. Download & Install next sequential build 20240522-52a73fc
22. Update
23. Open
24. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
25. Confirm notification from Omnipod 5 app
26. Download & Install next sequential build 20240528-c67b74f
28. Update
29. Open
30. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
31. Confirm notification from Omnipod 5 app
32. Download & Install next sequential build 20240529-27de713
33. Update
34. Open
35. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
36. Confirm notification from Omnipod 5 app
37. Download & Install next sequential build 20240601-a6fdfc4
38. Update
39. Open
40. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
41. Confirm notification from Omnipod 5 app

42. Download & Install next sequential build 20240611-1958827
43. Update
44. Open
45. 127.0.0.1:17580/pebble accesed via browser
Iob = "unknown"
46. Restart mobile device
47. 127.0.0.1:17580/pebble accesed via browser
Iob = "unknown"
48. Clear cache in browser
49. Restart mobile device
50.  127.0.0.1:17580/pebble accesed via browser
Iob = "unknown"

Identify build 20240611 does not function as expected

Attempt to go back to prior build 20240601-a6fdfc4

51. Download & Install next sequential build 20240601-a6fdfc4
52. Update
53. Open
54. 127.0.0.1:17580/pebble accesed via browser
Iob = "unknown"
55. Confirm notification from Omnipod 5 app

Attemts to troubleshoot

56. Clear cache in xDrip app
57. 127.0.0.1:17580/pebble accesed via browser
Iob = "unknown"

Conclusion
Something is broken.

Next Steps.
As i enjoy having the IoB functionality i am going to delete all xDrip
packages and anything else that rmay remain.

I will install the 20240419 build and use the functionality.

What I am missing is the ability to be able to easily share this data with
my spouse using the new xDrip Sync feature

I hope this helps.
Let me know if you need further clarifications, testing or insight.

Graham

Screenshots of initial setup and steps are available but i dont think they
will help with a resolution

**Reply by Navid200** [reply]
>....
> 42. Download & Install next sequential build 20240611-1958827
> 43. Update  

What is step 43?

After you install, xDrip is updated.
You can check the classic status page to confirm.  
I'm not sure what you mean by update after you have downloaded and installed already.

**Comment by gdcwatchfaces** [comment]
Hi Navid...

Just curious if you have any comments or thoughts on where the issue may
stem from?

**Reply by Navid200** [reply]
I can answer that if you identify the very first release of xDrip that caused this problem.

**Comment by gdc27health** [comment]
What?

Navid....
I do not understand what more you need from me.  I have given you 50+
steps that I have used to identify that xDrip+ breaks.  Do you think I am
just making this up?  What more do you want me to do?.

I do not do social media.  You may want to see in your facebook group if
others have experience

**Reply by Navid200** [reply]
@gdc27health 
I responded to the person who opened this thread.
You have been posting in this same thread.  As a result, you have been subscribed to the thread.  You can remove yourself.  When you are subscribed to a thread, you will receive a notification when someone posts in the thread.  You can clearly see I was not addressing you.

However, the message is the same.
I need to see exactly which version of xDrip was the very first version that caused this problem.  And posting a 50-line post is not making it a better post.  A short 2-lint post containing what matters is the best post.

If you install two xDrip versions and one works and other does not and then you install the same two in the opposite order and you see different outcome, that is not a reliable set of tests.

I don't know how else to explain this.  
I am a volunteer.  I do what I can to help improve xDrip.  Please don't make this personal.  If you are not happy with my style, that's OK.  Not everyone works the same way.  Please wait for someone else to address the problem.

If you make personal comments, like your last post, I will just stop interacting with you.  

If you don't understand what I mean by unreliable tests, please ask exactly that without making it sound like an offensive remark.  I will be happy to have a conversation and explain anything that is not clear.
Make another personal attack and I may even block you.

**Comment by gdc27health** [comment]
Navid

Do you not understand
gdcHealth
gdcWatchFaces

I am the same person.  It just comes to both of my emails.

I have given you everything you asked for.

You wanted to know if I installed Build A then Build B and it broke.  Could
I then "update" back to Build A

I think I have given you all of the information that you asked for.  If
that's not the case let me know what more you need from me so that I can
try and point you in the right direction in identifying the issue with the
xDrip software.

I tell you that it is repeatable.  I can do it over and over again and
break the code.  The only build i can reliably install it the April build

Is there anybody else at xDrip development that has looked at or chimed in
on this issue?

Bless your heart
Graham G in GDC

**Reply by Navid200** [reply]
These are holiday times.  Many are going to be away for a long time during this period.

This is a screenshot from one of your posts:  
![Screenshot 2024-12-10 164035](https://github.com/user-attachments/assets/79edbed2-3b7a-4acd-ab93-a60029e25707)

What does  "Update" mean?
Considering in 9, you say "Download and install".

If you download and install, you are done.  There is no need to update.  
What do you mean by "update"?  

This is the second time I am asking this question, which was never answered the first time.

**Comment by gdcwatchfaces** [comment]
Navid i respond to you with a screenshot of what the user is presented when
the software is installed.  It prompts the user to "Update"

**Reply by Navid200** [reply]
OK, I understand now.

Can you just install the two consecutive releases without clearing cache and without restoring settings?
Can you just install A and does it work?  And can you just install B?  Does it break?  If yes, will it work again if you just install A without anything else?

If yes, can you identify the two releases in a single line please?

**Comment by gdcwatchfaces** [comment]
Navid...

OK, I understand now.

Can you just install the two consecutive releases without clearing cache
and without restoring settings?

You can see in the steps I took to break the software.  1.  I never
restored the settings.  I just installed the next build right on top of the
previous build.  2. I only cleared the cache at the very end when it broke
to see if that was perhaps an issue.  That is a normal step in debugging.

Can you just install A and does it work?

As stated before . If i install Build A - the one I identify as the April
build.  Yes I can get it to work on a clean install

And can you just install B? Does it break?

I have tried to install many different 'just" Build B's and None of them
work

If yes, will it work again if you just install A without anything else?

 Yes I can only get it to work on a clean install

**Reply by Navid200** [reply]
No, A cannot be the April release.  A and B must be consecutive.
So, I take it you cannot find two consecutive releases that by just going back from B to A it would work again.

I will label this and it will have to wait for someone more familiar with pebble endpoint to address it.
Please be patient.
Thanks for reporting this.

**Comment by gdcwatchfaces** [comment]
Navid
What are your not understanding?\

IOB testing steps

1. All instances of xDrip and related files deleted
2. Mobile device restarted
3. xDrip build 2024041914 installed

4. DexShare configured with user id / password and us server
5. Settings > Inter-app stettings

   - Broadcast locally = ON
   - Noise Blocking > Send even Extremely noisy signals
   - Compatible Broadcast = checked
   - Identify receiver = de.michelinside.glucodatahandler
   - xDrip Web Service = ON
   - Broadcast Service API = ON
   - Fetch IoB from companion a.. = ON
   - Please Allow Permission = OK
   - Notification access slider to ON
   - Allow xDrip+ = Allow


6. Mobile device restarted
7.  127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
8. Confirm notification from Omnipod 5 app
9. Download & Install next sequential build 20240512-db904ab
10.Update
11. Open
12. Please Allow Permission = OK
13. Stop optimizing battery usage = Allow
14. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
15. Confirm notification from Omnipod 5 app
16. Download & Install next sequential build 20240515-c967ee7
17. Update
18. Open
19. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.80"
20. Confirm notification from Omnipod 5 app
21. Download & Install next sequential build 20240522-52a73fc
22. Update
23. Open
24. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
25. Confirm notification from Omnipod 5 app
26. Download & Install next sequential build 20240528-c67b74f
28. Update
29. Open
30. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
31. Confirm notification from Omnipod 5 app
32. Download & Install next sequential build 20240529-27de713
33. Update
34. Open
35. 127.0.0.1:17580/pebble accesed via browser
Iob = present with value "0.85"
36. Confirm notification from Omnipod 5 app
37. Download & Install next sequential build *20240601-a6fdfc4  BUILD A*
38. Update
39. Open
40. 127.0.0.1:17580/pebble accessed via browser
Iob = present with value "0.85"
41. Confirm notification from Omnipod 5 app

42. Download & Install next sequential build* 20240611-1958827 BUILD B*
43. Update
44. Open
45. 127.0.0.1:17580/pebble accessed via browser
Iob = "unknown"
46. Restart mobile device
47. 127.0.0.1:17580/pebble accessed via browser
Iob = "unknown"
48. Clear cache in browser
49. Restart mobile device
50.  127.0.0.1:17580/pebble accessed via browser
Iob = "unknown"

Identify build 20240611 does not function as expected

*These are consecutive Builds if i am not mistaken???:???*

*I will label this and it will have to wait for someone more familiar with
pebble endpoint to address it.*
You are just now telling me after i have put a dozen hours into debugging
this issue that you are not familiar?
How do we identify that person that is more familiar

Bless Your Heart Navid

Graham

**Reply by Navid200** [reply]
There is no way to identify anyone.
xDrip is driven by volunteers.  
If a volunteer can contribute and chooses to contribute, they will.  
This is the nature of open source.

I never asked you to spend a lot of time.  I offered to help if I could.  It turns out I cannot.
I hope you will find a solution.

**Comment by gdcwatchfaces** [comment]
You hope that I will find a solution?????????  I tried last night to follow
your directions on setting up a Nightscout account on Google Cloud.  I was
hoping that if i had a version of xDrip that was passing IOB that i could
pass it to nightscout
However the instructions that i followed 3 times resulted in updates being
needed to the virtual machine in some command line that were not
described in your instructions.  But I thank you anyway.

I am not a JAVA developer.  I create watch faces for the diabetic
community.  Like you I volunteer to do this.  Insulin On Board is a
significant piece of the diabetes puzzle

But thank you so much for your volunteering!!

Bless your heart
